scripts/m18_submission_checker/aida_m18_submission_checker.py

```diff
@@ -0,0 +1,295 @@
+
+def do_TA3_check(path_items):
+    """
+    Check the path in the archive to ensure the path is valid
+    Task 3 directory structure:
+    <TA1performer>_<run>(-<TA1performer>_<run>).<TA2performer>_<run>(-<TA2performer>_<run>).<TA3performer>_<run>
+        <TA1performer>_<run>(-<TA1performer>_<run>).<TA2performer>_<run>(-<TA2performer>_<run>).<TA3performer>_<run>.<SIN ID>.<SIN frame ID>.<H followed by three digits>.ttl			(1 to X)
+
+    Ignores non ttl files.
+    Prints number of ttl files found, and number of ttl files that are in need to be checked.
+
+    we do not check for all NIST standards set by the document, only for basic directory structure.
+    :param path_items: The path to the .ttl file that needs to be checked
+    :return: True if the file path is valid, False otherwise
+    :rtype: bool
+    """
+
+    # validate that the ttl file is in an acceptable directory
+    # rule: ttl needs to be in 2nd level -- aka 3rd item in list['name', 'NIST', 'valid.ttl']
+    # rule: ttl needs to be in directory named NIST, or INTER-TA
+
+    return True if len(path_items) == 2 else False
+
+
+
+def main():
+    """
+    :returns: True if script completed successfully (regardless if archive is submit ready or not)
+              False if script ran into a fatal error during execution and did not complete successfully.
+    :rtype: bool
+    """
+
+    # set logging to log/print INFO+ to stdout
+    logging.basicConfig(level='INFO', format='%(levelname)s:  %(message)s')
+
+    if check_runtime_args(sys.argv) is False:
+        return False
+
+    archive_file_name = str(sys.argv[1])
+
+    archive_file_type = get_valid_archive_filetype(archive_file_name)
+
+    if archive_file_type is None:
+        return False
+
+    archive_member_names = get_archive_member_names(archive_file_name, archive_file_type)
+
+    if archive_member_names is None:
+        return False
+
+    task_type = get_task_type(archive_file_name, archive_member_names)
+    if task_type == None:
+        return False
+
+    logging.info("Archive " + str(archive_file_name) + " is detected to be of task type " + task_type + ".")
+
+    archive_status = get_archive_submit_ready_status_values(task_type, archive_member_names)
+
+    # log archive report
+    logging.info("M18 Submission Checker Report for: \n\t{0}".format(archive_file_name))
+    logging.info("Total .ttl files found in submission archive: {0}".format(archive_status['total']))
+    logging.info("Total number of .ttl files to be validated: {0}".format(archive_status['valid']))
+    #if they have ttls located in invalid locations, print the warning
+    if archive_status['invalid'] > 0:
+        logging.warning("Total number of .ttl files located in invalid locations and will not be validated: {0}".format(archive_status['invalid']))
+        logger_print_level = logging.warning if archive_status['valid'] > 0 else logging.error
+        logger_print_level(warnings_dict[task_type])
+    if archive_status['valid'] > 0:
```

Comment:
The SUCCESS message should only be shown if archive_status['valid'] > 0 and archive_status['invalid'] == 0 (in other words, there are no TTLs in unexpected locations).

There should be one additional case:
if archive_status['valid'] > 0 and archive_status['invalid'] > 0:
log a warning
"PARTIAL SUCCESS! {0} is organized according to task {1} rules and can be submitted but there are .ttl files in unexpected locations that will not be validated"

Reply:
will write out diagram of output flow in next commit